### DISCUSSION TITLE
Please call out breaking changes better in the documentation

### DISCUSSION THREAD

**Discussion by balaji-dutt** [bpg/terraform-provider-proxmox#657] [General]
Hi,

I was struggling for quite a while today to figure out why my Terraform config using the provider abruptly stopped working with a "tls: protocol version not supported" error.

I eventually figured out it was due to [this "bug fix"](https://github.com/bpg/terraform-provider-proxmox/blob/main/CHANGELOG.md#bug-fixes-4) in 0.33.0 where the min. version was set to TLS 1.3. This is a breaking change for anyone using nginx on Debian 11 to load balance across clusters, as the version of nginx on Debian 11 does not support TLS 1.3 at all. 

It would be great if those sort of changes could be called out better in the documentation.

**Comment by bpg-dev** [comment]
Hi @balaji-dutt,

Thanks for your feedback!

I agree that we should have done a better job explaining this change – it's not your typical "bugfix", for sure.

As for whether it's a real "breaking change" that's a bit of a gray area. TLS 1.3 has been supported by PVE for a while now, and as of version 7.2, you can even set it to use TLS 1.3 exclusively. I get how frustrating it can be when an update messes with a perfectly working environment, but for most users, this won't be a "breaking change", especially if they're running standard home lab setups.

Your feedback is valuable, and we'll make sure to enhance our documentation to avoid these kinds of surprises in the future. If you have more questions or concerns, don't hesitate to ask!

**Reply by balaji-dutt** [reply]
Hi @bpg 

> As for whether it's a real "breaking change" that's a bit of a gray area. TLS 1.3 has been supported by PVE for a while now, and as of version 7.2, you can even set it to use TLS 1.3 exclusively. I get how frustrating it can be when an update messes with a perfectly working environment, but for most users, this won't be a "breaking change", especially if they're running standard home lab setups.

I agree that PVE has supported TLS 1.3 for a while now, but there is no requirement on the PVE side to limit connections to TLS 1.3 only by default. I would argue in a homelab setup it is less likely that folks will move to a more restrictive setup, compared to a full production environment. So there is really no specific reason that people who have expected/prepared for such a change unless it was called out explicitly somewhere.

Thank you for agreeing to looking into calling out such changes better in the documentation. Since the code seems to follow the Conventional Commits standard along with a changelog, perhaps you could consider adding the [Breaking Changes Convention](https://www.conventionalcommits.org/en/v1.0.0/#summary:~:text=in%20Semantic%20Versioning%29.-%2CBREAKING%20CHANGE%2C-%3A%20a%20commit) notation to such commits?
